src/state.rs

```diff
@@ -365,78 +370,126 @@ impl TabExpandedString {
     }
 }
 
-/// Estimate the number of seconds per step
+/// Estimate the number of steps per second
+///
+/// Exponentially smoothed estimate with an adjustable weighting
 ///
-/// Ring buffer with constant capacity. Used by `ProgressBar`s to display `{eta}`,
-/// `{eta_precise}`, and `{*_per_sec}`.
+/// The weight used for smoothing can be changed with `set_exponent_weight_secs`. All data in the
+/// Estimator older than this value will receive 10% of the weight in the estimate. Data newer than
+/// this value will receive 90% of the weight. The default value is 15 seconds.
 pub(crate) struct Estimator {
-    steps: [f64; 16],
-    pos: u8,
-    full: bool,
+    smoothed_steps_per_sec: f64,
+    double_smoothed_steps_per_sec: f64,
     prev: (u64, Instant),
+    start: Instant,
+    exponent_weight_seconds: f64,
+    double_smoothing: bool,
```

Comment:
That sounds fine to me. Though as some users in the comments pointed out, the behavior might become quite bad with rare events (on the order of an event every 30 seconds, say), because the smoothing would be happening over a (relatively) short period of time. 

I was also assuming some users would prefer single smoothing over double. But agreed on the importance of good defaults.